### PR TITLE
Add path to file watcher error & disallow empty paths

### PR DESCRIPTION
**Pull request by konradreiche**
I ran into this issue myself or have seen other run into it. Adding a path to the file watcher error message and disallowing empty paths to begin with could help give more directions when setting these things up or debugging them.

### CODE REVIEW

**Review by fishy — APPROVED** [review]
This probably should be split into 2 commits? But I don't feel super strongly about that so :shrug:

File: filewatcher/filewatcher.go [hunk 1]
```diff
@@ -186,7 +186,7 @@ func New(ctx context.Context, cfg Config) (*Result, error) {
 		select {
 		default:
 		case <-ctx.Done():
-			return nil, fmt.Errorf("filewatcher: context cancelled while waiting for file to load. %w", ctx.Err())
+			return nil, fmt.Errorf("filewatcher: context cancelled while waiting for file under %s to load. %w", cfg.Path, ctx.Err())
```
Comment (pacejackson) [inline filewatcher/filewatcher.go:189]: nit, `%q` rather than `%s` will probably give a clearer message.

File: secrets/errors.go [hunk 2]
```diff
@@ -9,6 +9,9 @@ import (
 // encoding in the secrets.json file.
 var ErrInvalidEncoding = errors.New("secrets: invalid encoding, expected identity, base64 or empty")
 
+// ErrEmptySecretPath is returned when the key for a secret is empty.
+var ErrEmptySecretPath = errors.New("secrets: secret path cannot be empty")
```
Comment (pacejackson) [inline secrets/errors.go:13]: Maybe this should be "key" rather than "path"?  Path sounds like the path to the secrets file IMO.

Comment (konradreiche) [reply secrets/errors.go:13]: Agreed, as I was re-reading it I was thinking of the file path.

Comment (fishy) [reply secrets/errors.go:13]: we do call them "secret path" in some context. our terminology is confusing 😞

**Review by fishy — APPROVED** [review]

File: secrets/errors.go (hunk 2 above)
Comment (konradreiche) [reply secrets/errors.go]: I did rename the parameter of those methods too.

**Review by pacejackson — APPROVED** [review]

**Review by fishy — APPROVED** [review]

**Pull request merged**